src/t_zset.c

```diff
@@ -2257,26 +2257,13 @@ int zuiFind(zsetopsrc *op, zsetopval *val, double *score) {
         return 0;
 
     if (op->type == OBJ_SET) {
-        if (op->encoding == OBJ_ENCODING_INTSET) {
-            if (zuiLongLongFromValue(val) &&
-                intsetFind(op->subject->ptr,val->ell))
```

Comment:
yep, RM_ScanKey also forgot it, and it will hang if scan a listpack encoding set, updated it and added test to cover it

> maybe we better remove the #include "intset.h" from server.h?

I'm not sure about this, can you explain this a bit?

Reply:
i was hoping we can implement the code in module.c without intset specific APIS (using some wrappers from t_list.c).
and then, if we remove the include to intset.h, we'll get compilation errors if anyone is attempting to use an intset specific API (which is what caused the mess here).

i.e. can we use setTypeInitIterator instead of having listpack and intset specific handling?